visidata/loaders/npy.py

```diff
@@ -11,44 +12,79 @@ def open_npz(vd, p):
+        if self._transpose:
+            source = self.npy[:,None]
+        else:
+            source = self.npy
+
+        nrows = len(self.npy)
+
+        if self._matrix_enumerate:
+            source = list(list((chain((i,), row))) for i, row in enumerate(source))
+
+        yield from Progress(source, nrows)
+
 
     def reloadCols(self):
+        self._matrix_enumerate = bool(self.options.get('npy_matrix_enumerate'))
+
         self.columns = []
-        for i, (name, fmt, *shape) in enumerate(self.npy.dtype.descr):
-            if not name:
-                continue
-            if shape:
-                t = anytype
-            elif 'M' in fmt:
-                self.addColumn(Column(name, type=date, getter=lambda c,r,i=i: str(r[i])))
-                continue
-            elif 'i' in fmt or 'u' in fmt:
-                t = int
-            elif 'f' in fmt:
-                t = float
+        if len(self.npy.shape)==1:
+            self._transpose = not bool(self.npy.dtype.names)
+            for i, (colname, fmt, *shape) in enumerate(self.npy.dtype.descr):
+                if not colname:
+                    colname = f"col{i}"
+                ctype = _guess_type(shape, fmt)
+                if ctype=="time":
+                    self.addColumn(Column(colname, type=date, getter=lambda c,r,i=i: str(r[i])))
+                    continue
+                self.addColumn(ItemColumn(colname, i, type=ctype))
+        elif len(self.npy.shape)==2:
+            ncols = self.npy.shape[1]
+            ctype = _guess_type(None, self.npy.dtype.descr[0][1])
+
+            if self._matrix_enumerate:
+                self.addColumn(ItemColumn("row", 0, width=8, keycol=1, type=int), index=0)
+                for i in range(ncols):
+                    self.addColumn(ItemColumn(f'col{i}', i+1, width=8, type=ctype), index=i+1)
             else:
-                t = anytype
-            self.addColumn(ColumnItem(name, i, type=t))
-
+                for i in range(ncols):
+                    self.addColumn(ItemColumn('', i, width=8, type=ctype), index=i)
+        else:
+            vd.fail(f"too many dimensions in shape {self.npy.shape}")
+
+def _guess_type(shape, fmt):
+    if shape:
+        return anytype
+    elif 'M' in fmt:
+        return "time"
+    elif 'i' in fmt or 'u' in fmt:
+        return int
+    elif 'f' in fmt:
+        return float
+    return anytype
 
 class NpzSheet(vd.ZipSheet):
     # rowdef: tuple(tablename, table)
     columns = [
-        ColumnItem('name', 0),
-        ColumnItem('length', 1, type=vlen),
+        ItemColumn('name', 0),
+        ItemColumn('length', 1, type=vlen),
     ]
 
     def iterload(self):
         numpy = vd.importExternal('numpy')
-        self.npz = numpy.load(str(self.source), encoding='bytes', **self.options.getall('npy_'))
+        self.npz = numpy.load(str(self.source), encoding='bytes', allow_pickle=bool(self.options.get('npy_allow_pickle')))
```

Comment:
Prefer `self.options.npy_allow_pickle` to `self.options.get(...)`

Reply:
fixed this and other similar cases